pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/pluginstep/InitContainerV2Step.java

```
@@ -104,16 +116,31 @@ public TaskRequest obtainTaskAfterRbac(
         stepParameters, ambiance, logPrefix, stepParameters.getStepGroupIdentifier());
 
     String stageId = ambiance.getStageExecutionId();
-    List<TaskSelector> taskSelectors = new ArrayList<>();
     consumeExecutionConfig(ambiance);
     initialiseTaskUtils.constructStageDetails(
         ambiance, stepParameters.getIdentifier(), stepParameters.getName(), StepOutcomeGroup.STEP_GROUP.name());
 
     TaskData taskData = initialiseTaskUtils.getTaskData(buildSetupTaskParams);
+    List<TaskSelector> taskSelectors = getTaskSelectors(ambiance, stepParameters);
     return TaskRequestsUtils.prepareTaskRequest(ambiance, taskData, referenceFalseKryoSerializer,
         TaskCategory.DELEGATE_TASK_V2, null, true, TaskType.valueOf(taskData.getTaskType()).getDisplayName(),
         taskSelectors, Scope.PROJECT, EnvironmentType.ALL, false, new ArrayList<>(), false, stageId);
   }
+
+  @NotNull
+  private ArrayList<TaskSelector> getTaskSelectors(Ambiance ambiance, InitContainerV2StepInfo stepParameters) {
+    if (!featureFlagService.isEnabled(
+            AmbianceUtils.getAccountId(ambiance), FeatureName.CD_CONTAINER_STEP_DELEGATE_SELECTOR)
+        || KUBERNETES_DIRECT != stepParameters.getInfrastructure().getType()) {
```

Should we throw an exception if the infrastructure type is not KUBERNETES_DIRECT?

---

if the FF is not enabled then we should not throw exception.

---

Let's discuss do we need CD_CONTAINER_STEP_DELEGATE_SELECTOR FF here and where delegate selectors from step groups are applied.